lib/internal/assert/myers_diff.js

```diff
@@ -0,0 +1,159 @@
+        return backtrack(trace, actual, expected, checkCommaDisparity);
+      }
+    }
+  }
+}
+
+function backtrack(trace, actual, expected, checkCommaDisparity) {
+  const actualLength = actual.length;
+  const expectedLength = expected.length;
+  const max = actualLength + expectedLength;
+
+  let x = actualLength;
+  let y = expectedLength;
+  const result = [];
+
+  for (let diffLevel = trace.length - 1; diffLevel >= 0; diffLevel--) {
+    const v = trace[diffLevel];
+    const diagonalIndex = x - y;
+    let prevDiagonalIndex;
+
+    if (diagonalIndex === -diffLevel ||
+      (diagonalIndex !== diffLevel && v[diagonalIndex - 1 + max] < v[diagonalIndex + 1 + max])) {
+      prevDiagonalIndex = diagonalIndex + 1;
+    } else {
+      prevDiagonalIndex = diagonalIndex - 1;
+    }
+
+    const prevX = v[prevDiagonalIndex + max];
+    const prevY = prevX - prevDiagonalIndex;
+
+    while (x > prevX && y > prevY) {
+      const value = !checkCommaDisparity ||
+        StringPrototypeEndsWith(actual[x - 1], ',') ? actual[x - 1] : expected[y - 1];
+      ArrayPrototypePush(result, { __proto__: null, type: 'nop', value });
+      x--;
+      y--;
+    }
+
+    if (diffLevel > 0) {
+      if (x > prevX) {
+        ArrayPrototypePush(result, { __proto__: null, type: 'insert', value: actual[x - 1] });
+        x--;
+      } else {
+        ArrayPrototypePush(result, { __proto__: null, type: 'delete', value: expected[y - 1] });
+        y--;
+      }
+    }
+  }
+
+  return result.reverse();
+}
+
+function printSimpleMyersDiff(diff) {
+  let message = [];
+
+  for (let diffIdx = 0; diffIdx < diff.length; diffIdx++) {
+    const { type, value } = diff[diffIdx];
+    if (type === 'insert') {
+      message += `${colors.green}${value}${colors.white}`;
+    } else if (type === 'delete') {
+      message += `${colors.red}${value}${colors.white}`;
+    } else {
+      message += `${colors.white}${value}${colors.white}`;
+    }
+  }
+
+  return `\n${message}`;
+}
+
+function printMyersDiff(diff, simple = false) {
+  const message = [];
+  let skipped = false;
+  let previousType = 'null';
+  let nopCount = 0;
+
+  for (let diffIdx = 0; diffIdx < diff.length; diffIdx++) {
+    const { type, value } = diff[diffIdx];
+    const typeChanged = previousType && (type !== previousType);
+
+    if (type === 'insert') {
+      ArrayPrototypePush(message, `${colors.green}+${colors.white} ${value}`);
+    } else if (type === 'delete') {
+      ArrayPrototypePush(message, `${colors.red}-${colors.white} ${value}`);
+    } else if (type === 'nop') {
+      if (nopCount <= kNopLinesToCollapse) {
+        ArrayPrototypePush(message, `${colors.white}  ${value}`);
+      }
+      nopCount++;
+    }
+
+    if (typeChanged && previousType === 'nop') {
+      if (nopCount > kNopLinesToCollapse) {
+        ArrayPrototypeSplice(message, message.length - 1, 0, `${colors.blue}...${colors.white}`);
+        skipped = true;
+      }
+      nopCount = 0;
+    }
```

Comment:
I believe we could optimize this a bit to not have to move any entries in the array and have a linear push.

Reply:
Yeah that's what I did in the very early implementation... I went for cleaner code instead of "less moving around", reverted now